echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+
+        # Evaluate whether `bokeh` has been globally initialized
+        if not DiagnosticPlot._is_bokeh_initialized:
+            # ---- Initialize the extension
+            hv.extension("bokeh")
+            # ---- Mark the class flag to prevent redundant initializations
+            DiagnosticPlot._is_bokeh_initialized = True
+
+        # Evaluate whether `bokeh.theme` has been globally initialized
+        if not DiagnosticPlot._is_theme_initialized:
+            # ---- Initialize the theme
+            hv.renderer("bokeh").theme = bok.themes.Theme(json=self._json_theme)
+
+        # Assign the data
+        self.data = survey
+
+        # Store the ESRI baselayer map to avoid redundant calls
+        self.tile = gv.tile_sources.EsriOceanBase
+
+    def mesh_cropping_results(self) -> pn.Column:
+
+        # Get the datasets
+        DATAINPUT = {
+            "Latitude/longitude": {
+                "cropped": (
+                    self.data.analysis["kriging"]["mesh_df"].copy()
+                    if "kriging" in self.data.analysis
+                    else {}
+                ),
+                "full": self.data.input["statistics"]["kriging"]["mesh_df"].copy(),
+            },
+            "Isobath-referenced": {
+                "cropped": self.data.analysis["kriging"]["mesh_df"].copy(),
+            },
+        }
+
+        # Get the isobath data
+        isobath_df = self.data.input["statistics"]["kriging"]["isobath_200m_df"]
+
+        # Amend the column names, if needed
+        [
+            v.rename(
+                columns={
+                    col: (
+                        "longitude"
+                        if "longitude" in col
+                        else "latitude" if "latitude" in col else col
+                    )
+                    for col in v.columns
+                },
+                inplace=True,
+            )
+            for k, v in DATAINPUT["Latitude/longitude"].items()
+        ]
+
+        # Initialize widget options
+        # ---- Variable options
+        var_options = ["Latitude/longitude", "Isobath-referenced"]
+        # ---- Get column names
+        var_columns = {
+            "Latitude/longitude": ["longitude", "latitude"],
+            "Isobath-referenced": ["x", "y"],
+        }
+        # ---- Tile options
+        var_tiles = {
+            "Latitude/longitude": self.tile,
+            "Isobath-referenced": None,
+        }
+        # ---- Axis options
+        var_axes = {
+            "Latitude/longitude": {
+                "xlab": "Longitude (\u00B0E)",
+                "ylab": "Latitude (\u00B0N)",
+            },
+            "Isobath-referenced": {
+                "xlab": "Normalized x",
+                "ylab": "Normalized y",
+            },
+        }
+
+        # Initialize `panel` extension
+        pn.extension()
+
+        # Create dropdown selector
+        var_dropdown = pn.widgets.Select(name="Variable", options=var_options)
```

Comment:
Seems better named as "Coordinate"
```suggestion
        # Create dropdown selector
        var_dropdown = pn.widgets.Select(name="Coordinate", options=var_options)
```